### PR TITLE
Added new "designer targets"

### PR DESCRIPTION
**Pull request by davkean**
These targets will end up containing all things related to designer scenarios, include properties that drive Solution Explorer and VS behavior, project rules (properties, items, etc).

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-analysis

File: src/Targets/Microsoft.CSharp.Designer.targets [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <AppDesignerFolder Condition="'$(AppDesignerFolder)' == ''">Properties</AppDesignerFolder>
+  
+    <!-- See: https://github.com/dotnet/roslyn-project-system/issues/201 -->
+    <CustomCommonXamlResourcesDirectory Condition="$(CustomCommonXamlResourcesDirectory) == ''">$(CommonXamlResourcesDirectory)</CustomCommonXamlResourcesDirectory>
+</PropertyGroup>
+
+  <ItemGroup>
+    <PropertyPageSchema Include="$(CustomCommonXamlResourcesDirectory)AppDesigner.xaml">
```
Comment (basoundr) [inline src/Targets/Microsoft.CSharp.Designer.targets]: Is a `\` required before `AppDesigner.xml`

Comment (davkean) [inline src/Targets/Microsoft.CSharp.Designer.targets]: No, directories, by convention have a trailing slash.

File: src/Targets/Microsoft.CSharp.Designer.targets [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  This file contains C# Visual Studio and designer-related properties and items for C# projects.
+  
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <AppDesignerFolder Condition="'$(AppDesignerFolder)' == ''">Properties</AppDesignerFolder>
+
```
Comment (basoundr) [inline src/Targets/Microsoft.CSharp.Designer.targets]: Could you please explain what happens when the condition `"'$(AppDesignerFolder)' == ''"` is true?

Comment (davkean) [inline src/Targets/Microsoft.CSharp.Designer.targets]: These are being documented here: https://github.com/dotnet/roslyn-project-system/blob/master/docs/well-known-project-properties.md

**Pull request merged**